packages/x-charts/src/ChartsReferenceLine/ChartsXReferenceLine.tsx

```diff
@@ -0,0 +1,96 @@
+import * as React from 'react';
+import { useDrawingArea, useXScale } from '../hooks';
+import {
+  CommonChartsReferenceLineProps,
+  ReferenceLineRoot,
+  getReferenceLineClasses,
+} from './common';
+import { ChartsText } from '../internals/components/ChartsText';
+
+export type ChartsXReferenceLineProps<
+  TValue extends string | number | Date = string | number | Date,
+> = CommonChartsReferenceLineProps & {
+  /**
+   * The x value associated to the reference line.
+   * If defined the reference line will be vertical.
+   */
+  x: TValue;
+};
+
+type GetTextPlacementParams = {
+  top: number;
+  height: number;
+} & Pick<CommonChartsReferenceLineProps, 'labelAlign'>;
+
+const getTextParams = ({ top, height, labelAlign = 'middle' }: GetTextPlacementParams) => {
+  switch (labelAlign) {
+    case 'start':
+      return {
+        y: top,
+        style: {
+          dominantBaseline: 'hanging',
+          textAnchor: 'start',
+        } as const,
+      };
+
+    case 'end':
+      return {
+        y: top + height,
+        style: {
+          dominantBaseline: 'auto',
+          textAnchor: 'start',
+        } as const,
+      };
+
+    default:
+      return {
+        y: top + height / 2,
+        style: {
+          dominantBaseline: 'central',
+          textAnchor: 'start',
+        } as const,
+      };
+  }
+};
+
+function ChartsXReferenceLine(props: ChartsXReferenceLineProps) {
+  const {
+    x,
+    label = '',
+    spacing = 5,
+    classes: classesProps,
+    labelAlign,
+    lineStyle,
+    labelStyle,
+    axisId,
+  } = props;
+
+  const { top, height } = useDrawingArea();
+  const xAxisScale = useXScale(axisId) as any;
```

Comment:
It has been more complicated than expected.

Each chart has some expectations about which scale they use, but `useXScale` returns any type of scale.

So the next line `xAxisScale(x)` was failing because `x` could be for example a string (available if using band scale) but the axis scale returned by `xAxisScale` could be linear which only accepts number

On the way of experimenting, I structured a bit more the scale types to remove as much `any` as possible in scales type def